### PR TITLE
Understanding the Basics of Responsive Web Design (RWD)

### LINKED ISSUE #1430: Understanding the Basics of Responsive Web Design (RWD)
**NOTE: We tend to stray away  or tend not to publish reviews/comparisons of commercial product offerings.**

### Brief Summary: 
Many global businesses are building websites to provide a platform for marketing and promoting their products. Building an attractive website only may not be enough. This is because customers require a site that enhances usability and responsiveness. Responsive web design ensures that the website responds well with the customers' devices, which leads to a great user experience. This article provides the basics of this concept in web development. 

### Key Takeaways: 
1. What is responsive web design (RWD)?
2. The building blocks of RWD
3. Why you need a responsive web design
4. How to get started with RWD
5. Examples of responsive web design
6. Conclusion

### References: 
Please list links to any published content/research that you intend to use to support/guide this article. (If none, please indicate 'N/A'.)

### PR DESCRIPTION
**Pull request by mbaabuones**
### Article Checklist
**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process, quality, and style-once published, you can then submit any number of articles.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #1430 .

### CODE REVIEW

File: articles/understanding-the-basics-of-responsive-web-design/index.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+The following are the main foundations of responsive web design.
+
+#### HTML and CSS 
+A combination of Hypertext Markup Language (HTML) and Cascading Styles Sheet (CSS) helps in controlling and adjusting the layout of a webpage. HTML controls the content, structure, and elements of a webpage. CSS technology edits the layout and design of elements that were created using HTML. 
+
+#### Fluid Grids
+This is an important principle of responsive design that uses relative measurement units (eg. percentages), rather than fixed units (like pixels). It enables the webpage to be resized to correspond with any screen size. With this component, a web page can adapt to the screen size or width of any device. 
+
+#### Media queries
+Media queries enhance specificity in responsive web design. They enable the website to adjust based on specific screen sizes. The website first uses the media queries to collect data that enables it to establish the screen size. The right CSS styles are then loaded to correspond to this screen size.
```
Comment (cyberShaw) [inline articles/understanding-the-basics-of-responsive-web-design/index.md:24]:
```suggestion
Media queries enhance specificity in responsive web design. They enable the website to adjust based on specific screen sizes. The website first uses media queries to collect data that enables it to establish the screen size. The right CSS styles are then loaded to correspond to this screen size.
```

File: articles/understanding-the-basics-of-responsive-web-design/index.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+**Step 2:** Choose a hosting plan
+There is a wide variety of hosting services for WordPress websites. Choose a hosting plan that corresponds to your financial capabilities and business needs. Click [here](https://websitesetup.org/best-wordpress-hosting-performance/#:~:text=10%20Best%20WordPress%20Hosting%20Services%201%20Bluehost%20%28www.Bluehost.com%29,Site5%20%28www.Site5.com%29%209%20DreamHost%20%28www.DreamHost.com%29%2010%20iPage%20%28www.iPage.com%29) to get a list of companies that can host your WordPress website.  
+
+**Step 3:** Images and content
+Prepare all the images and content that will be used on your website. Doing this in advance enables you to replace the demo content in your purchased theme with your ready content. 
+
+**Step 4:** Select a responsive WordPress Theme
+You can purchase a responsive WordPress theme [here](https://themeforest.net/popular_item/by_category?category=wordpress&_ga=2.256398347.1451571840.1610998232-961460759.1609948334). The theme should be selected based on the features, customization options, and the design. An appealing design should be chosen to attract customers to the responsive website.
```
Comment (cyberShaw) [inline articles/understanding-the-basics-of-responsive-web-design/index.md:52]:
```suggestion
You can purchase a responsive WordPress theme [here](https://themeforest.net/popular_item/by_category?category=wordpress&_ga=2.256398347.1451571840.1610998232-961460759.1609948334). The theme should be selected based on the features, customization options, and design. An appealing design should be chosen to attract customers to the responsive website. 
```

File: articles/understanding-the-basics-of-responsive-web-design/index.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+Responsive web design utilizes various elements that adjust the layout of a website to suit the orientation and screen size of a specific device (eg. fluid layouts, CSS codes, HTML5, and media queries). This enables users to interact with the website seamlessly. 
+
+### The Building Blocks of RWD
+The following are the main foundations of responsive web design.
+
+#### HTML and CSS 
+A combination of Hypertext Markup Language (HTML) and Cascading Styles Sheet (CSS) helps in controlling and adjusting the layout of a webpage. HTML controls the content, structure, and elements of a webpage. CSS technology edits the layout and design of elements that were created using HTML. 
+
+#### Fluid Grids
+This is an important principle of responsive design that uses relative measurement units (eg. percentages), rather than fixed units (like pixels). It enables the webpage to be resized to correspond with any screen size. With this component, a web page can adapt to the screen size or width of any device.
```
Comment (cyberShaw) [inline articles/understanding-the-basics-of-responsive-web-design/index.md:21]: You should mention some of these relative units like `vw` and `vh`. And also define their unit:

Example: 1 viewport width = 1/100th the width of the device.

File: articles/understanding-the-basics-of-responsive-web-design/index.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+**Step 5:** Install WordPress
+When you select a hosting plan, the hosting company will send a link to the email you used for registration. This link will take you to the hosting dashboard. When you fill in your credentials, you will find a section called ‘WordPress install’. Click on the WordPress icon and go through the instructions. 
+
+Another screen will pop up requiring you to fill in details such as your site’s description and desired credentials. After filling in all this information, click on ‘install’. Installing WordPress means that you can now access the WordPress dashboard using the registered credentials. 
+
+**Step 6:** Installing a theme
+In this step, you need to install the WordPress theme that you chose in step 4. In the WordPress dashboard, follow this selection sequence: **appearance**-> **themes**-> **add new**. Click on 'upload theme'. Click on 'install' to begin the installation. When this is done, click on 'activate'.
```
Comment (cyberShaw) [inline articles/understanding-the-basics-of-responsive-web-design/index.md:60]: You could attach screenshots here to support your instructions

File: articles/understanding-the-basics-of-responsive-web-design/index.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+#### 1. BMW
+BMW is a multinational automobile company that produces motorcycles and luxury vehicles. The website for this company consists of many background videos as well as images. The following is a screenshot of this website on a laptop. 
+
+![BMW via Laptop Browser](/engineering-education/understanding-the-basics-of-responsive-web-design/bmw-via-laptop-browser.png)
+
+All this content responds well to various screen sizes. The website appears as follows on a mobile device.
+
+![BMW via Mobile Browser](/engineering-education/understanding-the-basics-of-responsive-web-design/bmw-via-mobile-browser.jpg)
+
+You can access all the icons even with a mobile device. When using the laptop browser, the main icons appear at the top of the website. When using the mobile device, you can access these icons using a drop-down menu at the top left corner of the website.
```
Comment (cyberShaw) [inline articles/understanding-the-basics-of-responsive-web-design/index.md:83]: It is worth mentioning that it is called a Hamburger menu

**Review by cyberShaw — CHANGES_REQUESTED** [review]
Great article, @mbaabuones! I've suggested some minor changes to implement. Check those out.

File: articles/understanding-the-basics-of-responsive-web-design/index.md (hunk 1 above)
Comment (mbaabuones) [reply articles/understanding-the-basics-of-responsive-web-design/index.md]: Done.

File: articles/understanding-the-basics-of-responsive-web-design/index.md (hunk 2 above)
Comment (mbaabuones) [reply articles/understanding-the-basics-of-responsive-web-design/index.md]: Done.

File: articles/understanding-the-basics-of-responsive-web-design/index.md (hunk 3 above)
Comment (mbaabuones) [reply articles/understanding-the-basics-of-responsive-web-design/index.md]: Done.

File: articles/understanding-the-basics-of-responsive-web-design/index.md (hunk 4 above)
Comment (mbaabuones) [reply articles/understanding-the-basics-of-responsive-web-design/index.md]: Done.

File: articles/understanding-the-basics-of-responsive-web-design/index.md (hunk 5 above)
Comment (mbaabuones) [reply articles/understanding-the-basics-of-responsive-web-design/index.md]: Done.

**Comment by mbaabuones** [comment]
@cyberShaw I have finished working on the changes.

**Comment by cyberShaw** [comment]
I checked out the changes, @mbaabuones! They all look good. This article is ready for final review.

**Review by cyberShaw — APPROVED** [review]

**Comment by hectorkambow** [comment]
plagiarism check done

**Pull request merged**